### PR TITLE
Feedback PR

### PR DESCRIPTION
**Pull request by ImXron**
PR just for feedback. Will close afterward.

### CODE REVIEW

File: pytest_stress.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+import time
+
+SECONDS_IN_HOUR = 3700
```
Comment (blueyed) [inline pytest_stress.py]: Is this a nit to be found? ;)

File: pytest_stress.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            next_item = session.items[i + 1] if i + 1 < len(session.items) else None
+            item.config.hook.pytest_runtest_protocol(item=item, nextitem=next_item)
+            if session.shouldfail:
+                raise session.Failed(session.shouldfail)
+            if session.shouldstop:
+                raise session.Interrupted(session.shouldstop)
+        count += 1
+        if time.time() - start_time > _get_total_time(session):
+            break
+    return True
```
Comment (blueyed) [inline pytest_stress.py]: You could maybe also use `pytest_runtest_logfinish`, and set `session.shouldstop` therein then.
But if it works for you like that it's fine probably.. :)

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+sudo: false
```
Comment (hugovk) [inline .travis.yml]: `sudo:` no longer has any effect: https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

```suggestion
```

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+language: python
+
+matrix:
+    include:
+      - python: 2.7
+        env: TOX_ENV=py27
+      - python: 3.4
+        env: TOX_ENV=py34
```
Comment (hugovk) [inline .travis.yml]: Maybe drop EOL Python 3.4 and add the newly released 3.8?

File: README.rst [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+.. image:: https://img.shields.io/pypi/pyversions/pytest-stress.svg
+    :target: https://pypi.org/project/pytest-stress
+    :alt: Python versions
+
+.. image:: https://travis-ci.org/ImXron/pytest-stress.svg?branch=master
+    :target: https://travis-ci.org/ImXron/pytest-stress
+    :alt: See Build Status on Travis CI
+
+A plugin that allows you to loop tests for a user defined amount of time.
```
Comment (hugovk) [inline README.rst]:
```suggestion
A plugin that allows you to loop tests for a user-defined amount of time.
```

File: README.rst [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+Requirements
+------------
+
+Only tested with `Pytest`_ version 5.2.2.
+
+
+Installation
+------------
+
+You can install "pytest-stress" via `Pip`_ from `PyPI`_ (highly recommend installing in a `Pipenv`_)::
```
Comment (hugovk) [inline README.rst]:
```suggestion
You can install "pytest-stress" via `pip`_ from `PyPI`_ (highly recommend installing in a `Pipenv`_)::
```

File: README.rst [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+Loop tests for 1 hour 8 minutes and 9 seconds::
+
+    $ pytest --hours 1 --minutes 8 --seconds 9
+
+You can also add these values to config files::
+
+    [pytest]
+    addopts = --hours 1 --minutes 30
+
+Note: These loop times include setup and tear down operations as well. So if you have a test setup that takes 5
```
Comment (hugovk) [inline README.rst]:
```suggestion
Note: These loop times include setup and teardown operations as well. So if you have a test setup that takes 5
```

File: pytest_stress.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+def pytest_addoption(parser):
+    """
+    Add our command line options.
+    """
+    stress = parser.getgroup("stress")
+    stress.addoption(
+        "--hours",
+        action="store",
+        default=0,
+        help="The number of minutes to loop the tests for.",
```
Comment (hugovk) [inline pytest_stress.py]: ?
```suggestion
        help="The number of hours to loop the tests for.",
```

File: pytest_stress.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+        type=int,
+    )
+    stress.addoption(
+        "--minutes",
+        action="store",
+        default=0,
+        help="The number of hours to loop the tests for.",
```
Comment (hugovk) [inline pytest_stress.py]: ?
```suggestion
        help="The number of minutes to loop the tests for.",
```

File: pytest_stress.py [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+        type=int,
+    )
+    stress.addoption(
+        "--seconds",
+        action="store",
+        default=0,
+        help="The number of minutes to loop the tests for.",
```
Comment (hugovk) [inline pytest_stress.py]: ?
```suggestion
        help="The number of seconds to loop the tests for.",
```

File: pytest_stress.py [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+        type=int,
+    )
+
+
+def pytest_runtestloop(session):
+    """
+    Reimplement the test loop but loop for the user defined amount of time.
```
Comment (hugovk) [inline pytest_stress.py]:
```suggestion
    Reimplement the test loop but loop for the user-defined amount of time.
```

File: setup.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    author="Imran Mumtaz",
+    author_email="iomumtaz@gmail.com",
+    maintainer="Imran Mumtaz",
+    maintainer_email="iomumtaz@gmail.com",
+    license="MIT",
+    url="https://github.com/ImXron/pytest-stress",
+    description="A plugin that allows you to loop tests for a user defined amount of time.",
+    long_description=read("README.rst"),
+    py_modules=["pytest_pytest_stress"],
+    python_requires="!=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*",
```
Comment (hugovk) [inline setup.py]: aka
```suggestion
    python_requires=">=3.4",
```

Comment (hugovk) [inline setup.py]: Or if you drop EOL 3.4 (and remove the Trove classifier):
```suggestion
    python_requires=">=3.5",
```

File: tox.ini [hunk 13]
```diff
@@ -0,0 +1,12 @@
+# For more information about tox, see https://tox.readthedocs.io/en/latest/
+[tox]
+envlist = py27,py34,py35,py36,py37,pypy,flake8
```
Comment (hugovk) [inline tox.ini]: `setup.py` says only Python 3 is supported.

```suggestion
envlist =py34,py35,py36,py37,pypy3,flake8
```

File: .travis.yml [hunk 14]
```diff
@@ -0,0 +1,32 @@
+sudo: false
+language: python
+
+matrix:
+    include:
+      - python: 2.7
+        env: TOX_ENV=py27
```
Comment (hugovk) [inline .travis.yml]: `setup.py` says only Python 3 is supported

File: pytest_stress.py (hunk 1 above)
Comment (ImXron) [reply pytest_stress.py]: Lol the name you mean?

File: pytest_stress.py (hunk 2 above)
Comment (ImXron) [reply pytest_stress.py]: Ah good to know. Might be cleaner to break this functionality into multiple hooks. Will give it a shot thanks!

File: .travis.yml (hunk 3 above)
Comment (ImXron) [reply .travis.yml]: Thanks for all these comments Hugo! Will definitely these 👍

File: README.rst (hunk 7 above)
Comment (ImXron) [reply README.rst]: Lol my copy and paste fail 🤦‍♂

File: pytest_stress.py (hunk 1 above)
Comment (hugovk) [reply pytest_stress.py]: 60 seconds * 60 minutes = 3600, not 3700 :)

Comment (ImXron) [reply pytest_stress.py]: Oof! No wonder my hour long test was slightly longer 😂.

Good catch guys 👍.

**Pull request closed without merging**